src/cephadm/cephadm

```diff
@@ -4048,18 +4072,24 @@ def list_daemons(ctx, detail=True, legacy_dir=None):
                                     image_name = f.read().strip() or None
                             except IOError:
                                 pass
-                        i['container_id'] = container_id
-                        i['container_image_name'] = image_name
-                        i['container_image_id'] = image_id
-                        i['version'] = version
-                        i['started'] = start_stamp
-                        i['created'] = get_file_timestamp(
+                        i['container_id'] = container_id if container_id else ""
+                        i['container_image_name'] = \
+                            image_name if image_name else ""
+                        i['container_image_id'] = image_id if image_id else ""
+                        i['version'] = str(version)
+                        i['started'] = start_stamp if start_stamp else ""
+                        _created = get_file_timestamp(
```

Comment:
This change seems to be unrelated. Do you try to avoid a type error here?

Reply:
this is part of a commit to fix linting errors

Reply:
can you fix your linter here? The existing code is correct

Reply:
I'll need to check what was the error, but I'm pretty sure at the time it made sense.

Reply:
So, this is the error that is triggered when we don't account for the variable being possibly None.

![image](https://user-images.githubusercontent.com/506918/104276647-d6c41600-5494-11eb-9926-f19fcdb45efc.png)

Reply:
an alternative I think would be to set up asserts to ensure the variables are defined prior to assignment.